File: src/main/java/net/lyragames/practice/match/player/TeamMatchPlayer.java
package net.lyragames.practice.match.player;

import lombok.Getter;
import org.bukkit.Location;

import java.util.UUID;

/**
 * This Project is property of Zowpy © 2021
 * Redistribution of this Project is not allowed
 *
 * @author Zowpy
 * Created: 12/19/2021
 * Project: Practice
 */

@Getter
public class TeamMatchPlayer extends MatchPlayer {

    private final UUID teamUniqueId;

    public TeamMatchPlayer(UUID uuid, String name, Location spawn, UUID teamUniqueId) {
        super(uuid, name, spawn);
        this.teamUniqueId = teamUniqueId;
    }
}

File: src/main/kotlin/net/lyragames/practice/match/player/TeamMatchPlayer.kt
package net.lyragames.practice.match.player

import lombok.Getter
import org.bukkit.Location
import java.util.*

/**
 * This Project is property of Zowpy & EliteAres © 2021
 * Redistribution of this Project is not allowed
 *
 * @author Zowpy & EliteAres
 * Created: 12/19/2021
 * Project: Practice
 */
@Getter
class TeamMatchPlayer(uuid: UUID, name: String, spawn: Location, val teamUniqueId: UUID) :
    MatchPlayer(uuid, name, spawn)